resolver/blocking_resolver.go

```diff
@@ -371,8 +371,8 @@ func (r *BlockingResolver) handleDenylist(ctx context.Context, groupsToCheck []s
 		domain := util.ExtractDomain(question)
 		logger := logger.WithField("domain", domain)
 
-		if groups := r.matches(groupsToCheck, r.allowlistMatcher, domain); len(groups) > 0 {
-			logger.WithField("groups", groups).Debugf("domain is allowlisted")
+		if matches := r.matches(groupsToCheck, r.allowlistMatcher, domain); len(matches) > 0 {
+			logger.WithField("groups", maps.Keys(matches)).Debugf("domain is allowlisted")
```

Comment:
I think we can change this to log all the info:
```suggestion
			logger.WithField("matches", matches).Debug("domain is allowlisted")
```